src/AnonymousUser/AnonymousUserMiddleware.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace InsightArchitectures.AnonymousUser
```

Comment:
Root namespace should be: `InsightArchitectures.Extensions.AspNetCore`.
Also move this class into a Middleware to take it off from the root namespace.